sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreResponse.java

```diff
@@ -56,12 +60,38 @@ public StoreResponse(
         }
 
         this.status = status;
-        this.content = content;
-        if (this.content != null) {
-            this.responsePayloadLength = this.content.length;
+        replicaStatusList = new ArrayList<>();
+        if (contentStream != null) {
+            this.responsePayload = new JsonNodeStorePayload(contentStream, responsePayloadLength);
```

Comment:
if the response is not a valid json, then will the stream be closed as well? else the ByteBuf may never be released

Reply:
Fixed